src/GitHash.hs

```diff
@@ -162,6 +162,20 @@ getGitFilesRegular git = do
   let files3 = if packedExists then [packedRefs] else []
 
   return $ concat [files1, files2, files3]
+  where
+    charLF = fromIntegral $ fromEnum '\n'
+    charCR = fromIntegral $ fromEnum '\r'
+    -- this function can be implemented with Data.ByteString.dropWhileEnd,
```

Comment:
How old are we talking here? I'd rather have a simpler implementation here. I'm also not certain this is the logic we'd actually want. Would we want `takeWhile (not . isNewlineChar)`?

Reply:
`dropWhileEnd` appeared in `bytestring-0.10.12.0`. Since this package is constrained to `base >= 4.9.1.0`, i.e. `GHC >= 8.0.2` which results in `bytestring-0.10.8.1`, I considered using `dropWhileEnd` wouldn't be appropriate.

I assumed the content of the file conforms the RE
``` Regex
 (<git hash> | ref: <git ref>)[\n\r]*
```
Since, if I understand things right, no newline characters are allowed in git hash or git ref, the following two are equivalent.
``` Haskell
dropWhileEnd isNewlineChar
```
``` Haskell
takeWhile (not . isNewlineChar)
```
I chose the one with O(1) time complexity (assuming the trailing newlines are at most two bytes). However under the assumption that ref name is short, we could prefer brevity to runtime (We could say compile-time, since it's most frequently used in TH) cost.

Reply:
I'd be fine with either a tighter lower bound or the `takeWhile`. I'd prefer the `takeWhile` though. The performance difference shouldn't matter, and it sounds more resilient to format changes.

Reply:
OK. I'm gonna force-push the simpler expression (it'd be cumbersome to see a long function pops out in a commit and disappear in the next commit).

Reply:
@snoyberg Hey, I've been lately reading through `git help check-ref-format` and I was horrified to see that Non-ASCII characters are not prohibited in git references (Only bytes `0x00-0x1F` and `0x7F` are banned). Can I reimplement this with: 
``` Haskell
runGit ["symbolic-ref", "--quiet", "HEAD"]
```
following the advice in https://stackoverflow.com/questions/69174955/what-character-encoding-is-used-in-git-symbolic-refs-especially-on-windows ?

Reply:
I'm hesitant to say yes on this, since it can affect performance for all compilations, and it's not clear that there's a practical issue from this right now. Can you demonstrate this breaking in some way?

Reply:
Current implementation breaks if:
* `.git/HEAD` is in UTF-8, and
* the current branch contains Non-ASCII char (which is not prohibited by the rule; it accepts bytes >= 0x80).

So first I only thought using `ByteString` and `B8.unpack` would be bad.

Reply:
However we wouldn't like to use `text` for this. (Even if we used `text`, we wouldn't be able to read `.git/HEAD` unless we know the encoding. So I asked a question and they said it is unspecified.) I just thought we could assume `readCreateProcessWithExitCode` does things right on encodings.

Reply:
And I haven't seen a single git repo with non-ASCII branch name. I'd say anyone using such ref name is a unnamable monstrous entity coming from outside of sanity. However I've lost my sanity from fear once I saw the manpage.

Reply:
Let's ignore this issue for the moment, or maybe add a comment for it. It's always possible to address it in the future.

Reply:
You're correct. Easy things first.